src/PresentationalComponents/Inventory/_Inventory.scss

```diff
@@ -4,4 +4,22 @@
 
 .pf-v5-l-flex {
     --pf-v5-l-flex--spacer--column--base: var(--pf-v5-global--spacer--sm);
+}
+
+// Override Satellite/Foreman styles that impact the styling of the Inventory table
+// Prevent the toolbar buttons being pushed to the end of the toolbar
+.pf-v5-c-toolbar__group {
+    flex-grow: 0 !important;
+}
+.pf-v5-c-toolbar__group:first-child {
+    margin-right: 1rem !important;
+    width: auto !important;
```

Comment:
I see what you mean about ugly hack :P 

Is it possible to maybe add IDs to the components (or parents)?  Something to make the selectors more specific, so you wouldn't have to use `!important`. Happy to help from my side of the DOM too, if that's what you need.